gdal/alg/gdalwarper.cpp

```diff
@@ -477,11 +477,14 @@ GDALWarpNoDataMasker( void *pMaskFuncArg, int nBandCount, GDALDataType eType,
               {
                   if( ((bIsNoDataRealNan && CPLIsNan(padfWrk[iPixel*2])) ||
                        (!bIsNoDataRealNan &&
-                         ARE_REAL_EQUAL(padfWrk[iPixel*2], padfNoData[0])))
-                      && ((bIsNoDataImagNan && CPLIsNan(padfWrk[iPixel*2+1])) ||
-                          (!bIsNoDataImagNan &&
-                           ARE_REAL_EQUAL(padfWrk[iPixel*2+1],
-                                          padfNoData[1]))) )
+                         ARE_REAL_EQUAL(padfWrk[iPixel*2], padfNoData[0]))))
+
+// This part was commented out to conform to checking real part only for complex valued data
```

Comment:
The commented code should be removed, but warnings should be added in gdalwarper.h in the documentation of padfSrcNoDataImag and padfDstNoDataImag that they are used only for setting the nodata value, but that for comparing pixel values, only the real part is used.